src/andromede/simulation/benders_decomposed.py

```diff
@@ -208,54 +209,76 @@ def run(
 
 
 def build_benders_decomposed_problem(
-    network_on_tree: Dict[TreeNode, Network],
+    decision_tree_root: DecisionTreeNode,
     database: DataBase,
-    configured_tree: ConfiguredTree,
     *,
-    decision_coupling_model: Optional[Model] = None,
     border_management: BlockBorderManagement = BlockBorderManagement.CYCLE,
     solver_id: str = "GLOP",
+    coupling_network: Network = Network(""),
 ) -> BendersDecomposedProblem:
     """
     Entry point to build the xpansion problem for a time period
 
     Returns a Benders Decomposed problem
     """
 
-    master_network = create_master_network(network_on_tree, decision_coupling_model)
+    if not decision_tree_root.is_leaves_prob_sum_one():
+        raise RuntimeError("Decision tree must have leaves' probability sum equal one!")
+
+    null_time_block = TimeBlock(
+        0, [0]
+    )  # Not necessary for master, but list must be non-empty
+    null_scenario = 0  # Not necessary for master
 
-    # Benders Decomposed Master Problem
-    master = build_problem(
-        master_network,
+    coupler = build_problem(
+        coupling_network,
         database,
-        null_time_block := TimeBlock(  # Not necessary for master, but list must be non-empty
-            0, [0]
-        ),
-        null_scenario := 0,  # Not necessary for master
-        problem_name="master",
-        border_management=border_management,
+        null_time_block,
+        null_scenario,
+        problem_name="coupler",
         solver_id=solver_id,
-        problem_strategy=InvestmentProblemStrategy(),
+        build_strategy=InvestmentProblemStrategy(),
+        risk_strategy=ExpectedValue(0.0),
```

Comment:
This means that the objective is zero, right ? By doing so, if the user specify an objective contribution in the coupling model (which should not occur in practice...), the `build_problem` will create an objective with coefficient values equal to zero ? Could this cause issues for the solvers ?

Should we rather put a check in the coupling model to ensure that there is no objective ? But this means we are able to detect that some model is a coupling model within a tree, not easy...

Reply:
Yeah, that's right. The objective should be zero for the coupling model (at least for now?)
I don't know if coefficients 0 generate problem to the solvers, but we could check if a expression is 0 in the AST and skip it in the build problem ?